srcpkgs/mold/template

```diff
@@ -0,0 +1,49 @@
+# Template file for 'mold'
+pkgname=mold
+_commit_hash=191a69e36f5da5d18219ca5bce739c6e86039d31
+_ver=0.9.3-rc1
+version=0.9.3
+revision=1
+archs=x86_64*
+wrksrc=mold-${_ver}
+build_style=gnu-makefile
+makedepends="mimalloc-devel openssl-devel xxHash-devel tbb-devel zlib-devel"
+checkdepends="ripgrep libdwarf clang tar"
+short_desc="High performance drop-in replacement for existing Unix linkers"
+maintainer="Artur Sinila <freesoftware@logarithmus.dev>"
+license="AGPL-3.0-or-later"
+homepage="https://github.com/rui314/mold"
+changelog="https://github.com/rui314/mold/releases"
+distfiles="https://github.com/rui314/mold/archive/refs/tags/v${_ver}.tar.gz"
+checksum=ac35cdd14bcbd15e01b79831f50ce79f31830df225c1429725008f7eda322d7e
+
+# use system-wide installed mimalloc instead of the vendored one
+export SYSTEM_MIMALLOC=1
+export GIT_HASH=$_commit_hash
+
+if [ "$XBPS_TARGET_LIBC" = "glibc" ]; then
+	if [ "$XBPS_TARGET_WORDSIZE" -eq 64 ]; then
+		checkdepends+=" glibc-devel-32bit libgcc-devel-32bit"
```

Comment:
These packages only exist for `x86_64`, and we'd like to deprecate at some point. Just don't run `-m32` tests.

Reply:
@ericonr do you want to deprecate multilib altogether? Then how to run 32 bit programs, e. g. Steam?

Reply:
```sh
mkdir /r32
XBPS_ARCH=i686 xbps-install -S -r /r32 -R "$repo" base-system $packages
unlink /lib32
rm -rf /usr/lib32
ln -s /r32/lib /lib
ln -s /r32/lib /usr/lib32
```

Reply:
`mold` author disabled running `i386-*` tests for musl anyway

Reply:
@sgn sorry for chiming in like that, but what repo would I use for that small script?
When I use the multilib it cannot find the repodata, as it is `x86_64-repodata`.

Reply:
@Anachron: the main repository. Idea is to install regular i686 packages. Not sure how well it would work right now.

Reply:
@Chocimier indeed, I've done exactly that and I'm planning to test stuff (`wine`, `lutris`, `bottles` etc).
I'll probably post a new issue on void-packages with my results.

Reply:
My WIP can be found here: https://github.com/void-linux/void-packages/issues/34988